test/VASSAL/counters/StackTest.java

```diff
@@ -464,4 +466,83 @@ public void getPiecesReverseIteratorSholdIterateInReverseOrder() {
     }
 
   }
+
+  @Test
+  public void asListShouldReturnWithCorrectSize() {
+    // prepare
+    final GamePiece gamePiece1 = mock(GamePiece.class);
+    final GamePiece gamePiece2 = mock(GamePiece.class);
+    final GamePiece gamePiece3 = mock(GamePiece.class);
+
+    // run
+    Stack s = new Stack();
+    s.add(gamePiece1);
+    s.add(gamePiece2);
+    s.add(gamePiece3);
+    final List<GamePiece> pieces = s.asList();
+
+    // assert
+    assertEquals(3, pieces.size());
+  }
+
+  @Test
+  public void asListShouldGetAllPiecesInOrder() {
+    // prepare
+    final GamePiece gamePiece0 = mock(GamePiece.class);
+    final GamePiece gamePiece1 = mock(GamePiece.class);
+    final GamePiece gamePiece2 = mock(GamePiece.class);
+
+    // run
+    Stack s = new Stack();
+    s.add(gamePiece0);
+    s.add(gamePiece1);
+    s.add(gamePiece2);
+    final List<GamePiece> pieces = s.asList();
+
+    // assert
+    assertEquals(gamePiece0, pieces.get(0));
+    assertEquals(gamePiece1, pieces.get(1));
+    assertEquals(gamePiece2, pieces.get(2));
+  }
+
+  @Test
+  public void asListShouldReturnDefensiveCopy() {
```

Comment:
This test will fail with `asList()` returning an unmodifiable view.

Reply:
Test removed